Backpack/Classes/Panel/BPKPanel.m

```diff
@@ -0,0 +1,96 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "BPKPanel.h"
+
+#import <Backpack/Spacing.h>
+#import <Backpack/Color.h>
+#import <Backpack/Radii.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface BPKPanel()
+@property (assign) UIEdgeInsets originalLayoutMargins;
+
+- (void)setup;
+@end
+
+
+@implementation BPKPanel
+
+- (instancetype)initWithFrame:(CGRect)frame {
+    return [self initWithPadded:YES];
+}
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+
+    if (self) {
+        [self setup];
+    }
+
+    return self;
+}
+
+- (instancetype)initWithPadded:(BOOL)padded {
+    self = [super initWithFrame:CGRectZero];
+
+    if (self) {
+        [self setup];
+        self.padded = padded;
```

Comment:
Not sure it's an issue, but setup and self.padded will call the setPadded method twice during the runtime of this method. (Also, the order of calling setup and the padded setter matters.) 

Maybe it would be nicer to provide isPadded as a parameter of the setup method.
All other initialisers then can call setup with `[self setup:BPKPanelDefaultPaddedValue];`

Reply:
I like it